nexus/db-queries/src/db/datastore/address_lot.rs

```diff
@@ -53,12 +53,33 @@ impl DataStore {
             .transaction(&conn, |conn| async move {
                 let lot = AddressLot::new(&params.identity, params.kind.into());
 
-                let db_lot: AddressLot =
-                    diesel::insert_into(lot_dsl::address_lot)
-                        .values(lot)
-                        .returning(AddressLot::as_returning())
-                        .get_result_async(&conn)
-                        .await?;
+                // @internet-diglett says:
+                // I hate this. I know how to replace this transaction with
+                // CTEs but for the life of me I can't get it to work in
+                // diesel. I gave up and just extended the logic inside
+                // of the transaction instead chasing diesel trait bound errors.
+                let found_lot: Option<AddressLot> = lot_dsl::address_lot
+                    .filter(
+                        lot_dsl::name
+                            .eq(Name::from(params.identity.name.clone())),
+                    )
+                    .filter(lot_dsl::time_deleted.is_null())
+                    .select(AddressLot::as_select())
+                    .limit(1)
+                    .first_async(&conn)
+                    .await
+                    .ok();
+
+                let db_lot = match found_lot {
```

Comment:
In the context of a transaction, this "SELECT or else INSERT" behavior seems fine to me. "INSERT ... ON CONFLICT ... RETURNING" can really only tell you the value you *just* inserted anyway, so it'll omit any value which exists in the DB already.

(As you're already well-aware, CTEs are the right way to reduce this load with a smaller number of queries)